test/ExportVerilog/sv-dialect.mlir

```diff
@@ -64,11 +64,11 @@ rtl.module @M1(%clock : i1, %cond : i1, %val : i8) {
   // CHECK-NEXT:   else
```

Comment:
Since the reset block comes after the main block, I would put the reset block after the main block.

Also, it's generally a good idea to run the main block regardless of the reset then if the reset condition is met, run that. Doing this, you can reset only the registers about which you care and not require the compiler to create a feedback path (to retain the values) on all of the registers you don't care about. This helps FPGA mapping since not all registers are resettable. Not sure about ASIC, but presumably analogous optimizations could apply. I'd like to be able to express this behavior. If it's not a functional change, I would suggest it be the default so we produce more performant code by default.

Reply:
This is avoided by only generating assignments in the appropriate always block.  If you don't have a reset, you shouldn't be putting the assignment in a block with a reset.  Keeping separate always blocks for each type (and signal) of reset is strongly recommended practice (and expected by some tools).  The point of having the reset condition in the alwaysff block is precisely to make it easy to do this.

Reply:
There are some cases in which you want to have all the associated logic in the same always block. Common example is you generally want to include the data path and associated control path logic in the same block. You generally don't care about the data so long as your control logic gets reset. You want them in the same block for readability. It may also be necessary in some cases, but I'm not sure. This pattern works quite well in Quartus and is recommended for high-speed FPGA designs.